### PR TITLE
How to join the Butterfly-network

### LINKED ISSUE #271: Add documentation for the butterfly network
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated improvement to propose.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [X] Other

### Improvement Suggestion

Hi!

There is no documentation about `butterflynet`, while it is probably the most handy target for testing purpose.
It would nice to add it.
Where should I start?

### PR DESCRIPTION
**Pull request by rjan90**
A small knowledge base article about how one can join the Butterfly-network. Has a big warning that the network can not be considered stable, and may be reset or upgraded at any moment with little notice.

Fixes #271

### CODE REVIEW

File: content/en/kb/butterfly-network/index.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: "How to join the Butterfly network."
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:2]: Can we make it say Butterfly (test). We need to make is very clear that this is a test network.

File: content/en/kb/butterfly-network/index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+description: "A guide for how you can join the Butterfly network"
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:3]:
```suggestion
description: "A guide to join the Butterfly network"
```

File: content/en/kb/butterfly-network/index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+menu:
+  kb:
+    parent: "browse"
+toc: false
+pinned: false
+types: ["solution"]
+areas: ["Lotus Node", "Lotus Miner"]
+---
+
+## Problem
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:16]: We don't need to follow problem/solution format here. This is not a solution but an article. 
```suggestion
## Butterfly Network
```

File: content/en/kb/butterfly-network/index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+How to join the `Butterfly` network.
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:18]: We need to add full description here about butterfly net.
purpose
Use case
How to check notification for reset
Possible reset reasons
Why should you join it

File: content/en/kb/butterfly-network/index.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+## Resolution
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:20]:
```suggestion
## Steps to join butterfly network
```

File: content/en/kb/butterfly-network/index.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+description: "A guide for how you can join the Butterfly network"
+date: 2022-03-17T12:00:35+01:00
+lastmod: 2021-11-16T12:00:35+01:00
+draft: false
+menu:
+  kb:
+    parent: "browse"
+toc: false
+pinned: false
+types: ["solution"]
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:12]:
```suggestion
types: ["article"]
```

File: content/en/kb/butterfly-network/index.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+areas: ["Lotus Node", "Lotus Miner"]
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:13]:
```suggestion
areas: ["Butterfly Network", "Lotus Build"]
```

**Review by LexLuthr — CHANGES_REQUESTED** [review]

File: content/en/kb/butterfly-network/index.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    parent: "browse"
+toc: false
+pinned: false
+types: ["article"]
+areas: ["Butterfly Network", "Lotus Build"]
+---
+
+## Butterfly Network
+
+The `Butterfly` testnetwork is mostly an internal development and test network used by the Lotus team. Its usecase is mostly for short term testing, and should not be used for something that needs a long time operating network. To get updates about the Butterfly test network it´s recommended to follow the `#lotus-butterfly` channel in the Filecoin Slack, as well as subscribing to [network updates](https://status.filecoin.io).
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:18]:
```suggestion
The `Butterfly` test network is mostly an internal development and test network used by the Lotus team. It is used for short term testing, and should not be used for something that needs a long time operating network. To get updates about the Butterfly test network it is recommended to follow the `#lotus-butterfly` channel in the Filecoin Slack, as well as subscribing to [network updates](https://status.filecoin.io).
```

File: content/en/kb/butterfly-network/index.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+## Steps to join butterfly network
+
+{{< alert icon="warning" >}}
+The `Butterfly` network can not be considered a stable network and may be reset or upgraded at any moment with little notice.
+{{< /alert >}}
+
+Clone and checkout a release or branch as explained in the [build and install Lotus]({{< relref "../../lotus/install/linux/#build-and-install-lotus" >}}) guide.
+
+To join the `Butterfly` network, simply build Lotus by:
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:28]:
```suggestion
To join the `Butterfly` network, build Lotus with below commands:
```

File: content/en/kb/butterfly-network/index.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+```shell
+make clean
+make butterflynet
+make install
+```
+
+The `Butterfly` network supports three sector sizes, 512MiB, 32GiB and 64GiB.
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:35]:
```suggestion
```

File: content/en/kb/butterfly-network/index.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+toc: false
+pinned: false
+types: ["article"]
+areas: ["Butterfly Network", "Lotus Build"]
+---
+
+## Butterfly Network
+
+The `Butterfly` testnetwork is mostly an internal development and test network used by the Lotus team. Its usecase is mostly for short term testing, and should not be used for something that needs a long time operating network. To get updates about the Butterfly test network it´s recommended to follow the `#lotus-butterfly` channel in the Filecoin Slack, as well as subscribing to [network updates](https://status.filecoin.io).
+
```
Comment (LexLuthr) [inline content/en/kb/butterfly-network/index.md:19]:
```suggestion
The `Butterfly` network supports sector size of 512MiB, 32GiB and 64GiB.

```

File: content/en/kb/butterfly-network/index.md (hunk 4 above)
Comment (rjan90) [reply content/en/kb/butterfly-network/index.md]: I added description about the network/purpose/reset notifications/subscription.

 Not sure if we should add `possible reset reasons` and `Why should you join it` just yet, as it´s still mostly an internal development network that happens to be open to everybody. If we change it to a more open developer network at some point, I think we should add those points though! Wdyt?

**Review by LexLuthr — CHANGES_REQUESTED** [review]
Couple of more changes. Rest looks good.

File: content/en/kb/butterfly-network/index.md (hunk 4 above)
Comment (LexLuthr) [reply content/en/kb/butterfly-network/index.md]: Fair enough.

**Review by LexLuthr — APPROVED** [review]

**Pull request merged**